### DISCUSSION TITLE
default severity level for reportUnusedVariable in type checking mode basic

### DISCUSSION THREAD

**Discussion by excubo-jg** [microsoft/pyright#5951] [General]
I have moved a repo which tested without errors and warnings in basic mode to strict. From the initially 18k errors almost all  were caused by sloppy or missing annotations and knock-on effects. Though, the changed revealed a handful of bugs in the program logic all of which were caused by typos in variable names which triggered reportUnusedVariable-error messages in strict mode. I am wondering if it would be better to have warnings generated in basic mode as unused variables have a high likelihood of not reflecting the intention of the programmer.

**Comment by erictraut** [comment]
We are very conservative about which diagnostic rules to enable by default in basic mode because users do not like to see false positive errors or warnings. Most unused variables usually just represent old code — something that could be deleted but isn't. Most of these cases are therefore probably not real bugs, so reporting them as such (even as a warning) would be seen as a false positive.

You can always enable the `reportUnusedVariable` diagnostic rule manually if you feel that it is useful for your project.
